### PR TITLE
Added Unit Tests for Vector_Math, Flight_Boundary, Flight_Boundaries_Container, Obstacle, and Location 

### PR DESCRIPTION
**Pull request by Inffzy**

### CODE REVIEW

File: tests/test_flight_boundary.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from matplotlib.path import Path
+from SDA import*
+import unittest
+import numpy as np
```
Comment (DarkmatterVale) [inline tests/test_flight_boundary.py]: Please remove the ```np``` reference. I know this is done a lot in the codebase, but it is a bad habit...

File: tests/test_flight_boundary.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from matplotlib.path import Path
+from SDA import*
```
Comment (DarkmatterVale) [inline tests/test_flight_boundary.py]: Please fix the spacing for the import

File: tests/test_flight_boundary.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import unittest
+import numpy as np
+
+class FlightBoundaryTestCase(unittest.TestCase):
+
+    def setUp(self):
+
```
Comment (DarkmatterVale) [inline tests/test_flight_boundary.py]: Please remove this new line at the start of the method. Please do this for all methods

File: tests/test_flight_boundary.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        my_min_altitude = 50
```
Comment (DarkmatterVale) [inline tests/test_flight_boundary.py]: Please rename this. You should never use ```my``` in any name.

File: tests/test_flight_boundary.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        my_max_altitude = 200
```
Comment (DarkmatterVale) [inline tests/test_flight_boundary.py]: Same as the above comment. Please use more informative variables for all tests.

File: tests/test_polygon.py [hunk 6]
```diff
@@ -0,0 +1,178 @@
+from SyntheticDataset.image_operations import *
```
Comment (DarkmatterVale) [inline tests/test_polygon.py]: Please remove this file

File: tests/unit_tests/test_sda_flight_boundary.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+from matplotlib.path import Path
+from SDA import*
+import unittest
+import numpy as np
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_sda_flight_boundary.py]: Please remove the ```as np``` and leave it as ```numpy```

File: tests/unit_tests/test_sda_flight_boundary.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+class FlightBoundaryTestCase(unittest.TestCase):
+
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_sda_flight_boundary.py]: This is a very important class. Please add additional tests to ensure that the path works in a complex, irregular polygon. You need to test paths that will be edge cases (right outside of the path, right inside, in a concave place, etc)

File: tests/test_flight_boundary.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+    def test_is_point_in_bounds(self):
+
+        my_point = np.array([50, 50, 100])
+        self.assertTrue(my_flight_boundary.is_point_in_bounds(my_point), True)
+
+    def test_path_method(self):
+
+        my_point = np.array([50, 50, 100])
+        self.assertTrue(path_method(my_point), True)
```
Comment (DarkmatterVale) [inline tests/test_flight_boundary.py]: Please apply all of these comments to the other flight boundary test file, and delete this file.

File: tests/unit_tests/test_sda_vector_math.py [hunk 10]
```diff
@@ -0,0 +1,73 @@
+import math
+import numpy as np
+import unittest
+from SDA import VectorMath
+
+class VectorMathTestCase(unittest.TestCase):
+
+    def setUp(self):
+        pass
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_sda_vector_math.py]: Since you are passing in this method, you do not need to add a declaration

File: tests/unit_tests/test_suassystem_location.py [hunk 11]
```diff
@@ -0,0 +1,30 @@
+import unittest
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_suassystem_location.py]: Rod beat you to this test file...Please remove it

**Review by DarkmatterVale — CHANGES_REQUESTED** [review]
Overall, it looks good. You need to fix variable naming, spacing, and you need to remove duplicate files

File: tests/unit_tests/test_sda_flight_boundaries_container.py [hunk 12]
```diff
@@ -1,18 +1,18 @@
...
+        self.test_flight_boundaries_container = FlightBoundariesContainer(fly_zones)
 
     def test_is_point_in_bounds(self):
 
-        my_point1 = np.array([20, 30, 40])
-        self.assertTrue(self.my_flight_boundaries_container.is_point_in_bounds(my_point1), True)
-        my_point2 = np.array([1000, 1000, 1000])
-        self.assertFalse(self.my_flight_boundaries_container.is_point_in_bounds(my_point2), True)
+        test_point1 = numpy.array([20, 30, 40])
+        self.assertTrue(self.test_flight_boundaries_container.is_point_in_bounds(test_point1), True)
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_sda_flight_boundaries_container.py]: In an assertTrue call, you do not need to follow your expression with a True parameter. So, in this case, your call should be ```self.assertTrue(self.test_flight_boundaries_container.is_point_in_bounds(test_point1))```

File: tests/unit_tests/test_sda_flight_boundaries_container.py [hunk 13]
```diff
@@ -1,18 +1,18 @@
...
+        test_point2 = numpy.array([1000, 1000, 1000])
+        self.assertFalse(self.test_flight_boundaries_container.is_point_in_bounds(test_point2), True)
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_sda_flight_boundaries_container.py]: Same as the above comment.

File: tests/unit_tests/test_sda_flight_boundary.py [hunk 14]
```diff
@@ -1,25 +1,91 @@
...
+        test_max_altitude1 = 200.0
+
+        test_min_altitude2 = -300.0
+        test_max_altitude2 = -50.0
+
+        test_boundary_waypoints1 = numpy.array([(0,0),(0,100),(100,100),(100,0)])
+        test_boundary_waypoints2 = numpy.array([(-25,0),(50,50),(0,100),(-100,150),(100,150),(150,0),(50,-50),(0,-100),(100,-150),(-100,-150)])
+
+        test_bound_path1 = Path(test_boundary_waypoints1)
+        test_bound_path2 = Path(test_boundary_waypoints2)
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_sda_flight_boundary.py]: Lines 19 and 20 (the ```test_bound_path``` variables) are not used in the file and can be removed.

**Review by DarkmatterVale — CHANGES_REQUESTED** [review]
Almost perfect. Just a couple of changes remaining.

**Review by DarkmatterVale — APPROVED** [review]
